rust/template/cmd_parser/lib.rs

```diff
@@ -2,31 +2,55 @@
 
 mod parse;
 
-use std::io;
-use std::io::BufRead;
-use std::io::BufReader;
+use std::fs::{ self, File };
+use std::io::{ self, BufRead, BufReader };
 
 pub use parse::*;
 
 use nom::*;
 use rustyline::error::ReadlineError;
 use rustyline::Editor;
 
+use tokio::sync::{ mpsc, mpsc::error::TrySendError, oneshot };
+
 const HISTORY_FILE: &str = "cmd_parser_history.txt";
 
 // We handle stdin differently depending on whether it is a user terminal or a pipe.
 enum Input {
     Tty(Editor<()>),
     Pipe(BufReader<io::Stdin>),
+    Source(BufReader<fs::File>),
+    TempStore,
+}
+
+#[derive(Debug)]
+struct CliCmdReq {
+    cli_cmd: Vec<u8>,
+    resp: Option<oneshot::Sender<Result<bool, String>>>,
+}
+
+fn send_cmd_to_parser(tx: &mpsc::Sender<CliCmdReq>, cmd: CliCmdReq) -> Result<bool, String> {
```

Comment:
I think a better name would be `send_cmd_to_ddlog` or `send_command_to_executor`, since the receiver both parses and executes the command.